File: year2020/day01/java/Solution.java
import java.nio.file.*;
import java.util.*;

public class Solution {
    private static int[] NUMBERS;

    public static void main(String[] args) { 
        try {
            setup(Files.readString(Paths.get("../puzzle.txt")));
            System.out.println(part1());
            System.out.println(part2());
        } catch(Exception e) {
            e.printStackTrace();
        }
    }
    
	public static int part1() {
        for (int i = 0; i < NUMBERS.length - 1; i++) {
            for (int j = i + 1; j < NUMBERS.length; j++) {
                if (NUMBERS[i] + NUMBERS[j] == 2020) {
                    return NUMBERS[i] * NUMBERS[j];
                }
            }
        }
        return -1;
    }
    
    public static int part2() {
        for (int i = 0; i < NUMBERS.length; i++) {
            for (int j = i + 1; j < NUMBERS.length; j++) {
                for (int k = j + 1; k < NUMBERS.length; k++) {
                    if (NUMBERS[i] + NUMBERS[j] + NUMBERS[k] == 2020) {
                        return NUMBERS[i] * NUMBERS[j] * NUMBERS[k];
                    }
                }
            }
        }        
        return -1;
    }

    public static void setup(String input) {
        NUMBERS = Arrays.stream(input.split("\n")).mapToInt(Integer::parseInt).toArray();
    }
}

File: year2020/day01/python/solution.py
def main():
    print(part_1())
    print(part_2())


def part_1():
    for i in range(len(input) - 1):
        for j in range(i + 1, len(input)):
            if input[i] + input[j] == 2020:
                return input[i] * input[j]


def part_2():
    for i in range(len(input)):
        for j in range(i + 1, len(input)):
            for k in range(j + 1, len(input)):
                if input[i] + input[j] + input[k] == 2020:
                    return input[i] * input[j] * input[k]
                

def setup(path):
    global input
    
    with open(path, 'r') as file:
        input = [int(row) for row in file]


if __name__ == '__main__':
    setup('../puzzle.txt')
    main()